src/codewhisperer/client/codewhisperer.ts

```diff
@@ -39,8 +39,23 @@ export type ListCodeScanFindingsRequest = Readonly<CodeWhispererClient.ListCodeS
 export type ArtifactType = Readonly<CodeWhispererClient.ArtifactType>
 export type ArtifactMap = Readonly<CodeWhispererClient.ArtifactMap>
 export class DefaultCodeWhispererClient {
+    private credentials?: CognitoIdentityCredentials
+
     private async createSdkClient(): Promise<CodeWhispererClient> {
-        const credentials = !isCloud9() ? await getCognitoCredentials() : undefined
+        try {
+            if (!this.credentials && !isCloud9()) {
+                this.credentials = await getCognitoCredentials()
+            }
+            if (this.credentials && this.credentials?.needsRefresh()) {
```

Comment:
`this.credentials` is always true now?

Reply:
if it's Cloud9 it should still be undefined